py/server/deephaven/column.py

```diff
@@ -32,47 +32,118 @@ def __repr__(self):
         return self.name
 
 
-@dataclass
-class Column:
-    """ A Column object represents a column definition in a Deephaven Table. """
-    name: str
-    data_type: DType
-    component_type: DType = None
-    column_type: ColumnType = ColumnType.NORMAL
+class Column(JObjectWrapper):
+    """A Column object represents a column definition in a Deephaven Table."""
+
+    j_object_type = _JColumnDefinition
+
+    def __init__(
+        self,
+        name: str = None,
+        data_type: DType = None,
+        component_type: DType = None,
+        column_type: ColumnType = ColumnType.NORMAL,
+        j_column_definition: jpy.JType = None,
+    ):
+        if j_column_definition:
+            if name or data_type or component_type or column_type != ColumnType.NORMAL:
+                raise DHError(
+                    "Must specify the data fields or j_column_definition for Column, but not both."
+                )
+            self.name = j_column_definition.getName()
+            self.j_column_definition = j_column_definition
+        else:
+            if not name or not data_type:
+                raise DHError("Must specify both name and data_type")
+            if hasattr(data_type.j_type, "jclass"):
+                j_data_type = data_type.j_type.jclass
+            else:
+                j_data_type = data_type.qst_type.clazz()
+            j_component_type = (
+                component_type.qst_type.clazz() if component_type else None
+            )
+            j_column_type = column_type.value
+            self.name = name
+            self.j_column_definition = _JColumnDefinition.fromGenericType(
+                name, j_data_type, j_component_type, j_column_type
+            )
 
     @property
-    def j_column_header(self):
-        return _JColumnHeader.of(self.name, self.data_type.qst_type)
+    def j_object(self) -> jpy.JType:
+        return self.j_column_definition
+
+    @cached_property
+    def data_type(self) -> DType:
+        return from_jtype(self.j_column_definition.getDataType())
+
+    @cached_property
+    def component_type(self) -> DType:
+        return from_jtype(self.j_column_definition.getComponentType())
+
+    @cached_property
+    def column_type(self) -> ColumnType:
+        return ColumnType(self.j_column_definition.getColumnType())
 
     @property
-    def j_column_definition(self):
-        if hasattr(self.data_type.j_type, 'jclass'):
-            j_data_type = self.data_type.j_type.jclass
-        else:
-            j_data_type = self.data_type.qst_type.clazz()
-        j_component_type = self.component_type.qst_type.clazz() if self.component_type else None
-        j_column_type = self.column_type.value
-        return _JColumnDefinition.fromGenericType(self.name, j_data_type, j_component_type, j_column_type)
+    def j_column_header(self) -> jpy.JType:
+        return _JColumnHeader.of(self.name, self.data_type.qst_type)
```

Comment:
Since it is only used by `InputColumn`, remove this property and just call the Java factory method directly in `InputColumn._to_j_column`?